event_listener/dfusion_db/models.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import annotations
+from typing import NamedTuple, Dict, Any, List
+from enum import Enum
+
+class TransitionType(Enum):
+    Deposit = 0
+    Withdraw = 1
+    Auction = 2
+
+class StateTransition(NamedTuple):
+    transition_type: TransitionType
+    state_index: int
+    state_hash: str
+    slot: int
+
+    @classmethod
+    def fromDictionary(cls, data: Dict[str, Any]) -> StateTransition:
```

Comment:
Can we actually refer to a Class before it has been defined? My linter says Unresolved Reference.

Reply:
that's what the annotation import below is for